### PR TITLE
split up commands by type into traits that can be composed

### PR DESCRIPTION
**Pull request by 72squared**
this makes it easier to manage changes for each type of redis command set

### CODE REVIEW

File: modules/core/src/main/scala/com/github/scoquelin/arugula/commands/LettuceRedisAsyncCommands.scala [hunk 1]
```diff
@@ -230,24 +48,8 @@ private[commands] class LettuceRedisAsyncCommands[K, V](connection: RedisConnect
    * @tparam T The type of the returned value inside the io.lettuce.core.RedisFuture[T]
    * @return A Scala Future[T] extracted from the io.lettuce.core.RedisFuture[T]
    */
-  private def delegateRedisClusterCommandAndLift[T](command: JRedisClusterAsyncCommands[K, V] => RedisFuture[T]): Future[T] =
+  override private[commands] def delegateRedisClusterCommandAndLift[T](command: JRedisClusterAsyncCommands[K, V] => RedisFuture[T]): Future[T] =
```
Comment (72squared) [inline modules/core/src/main/scala/com/github/scoquelin/arugula/commands/LettuceRedisAsyncCommands.scala:51]: thought about renaming this to to just `delegate` or `delegateAndLift` for simplicity but we can do that in a subsequent MR. this is just about splitting up the functionality into a more modular design.

File: modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.github.scoquelin.arugula.commands.internal
```
Comment (72squared) [inline modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala:1]: one other thought ... I could place all of these into the `com.github.scoquelin.arugula.commands` namespace to keep it symmetrical with how it is organized in the api layer.

File: modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisCommandDelegation.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+   */
+  private[commands] def delegateRedisClusterCommandAndLift[T](
+    command: io.lettuce.core.cluster.api.async.RedisClusterAsyncCommands[K, V] => io.lettuce.core.RedisFuture[T]
+  ): Future[T]
+
+  /**
+   * The execution context
+   * Should be injected by the implementing class
+   */
+  private[commands] implicit val executionContext: ExecutionContext
```
Comment (72squared) [inline modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisCommandDelegation.scala:28]: I don't love how I have to define the `executionContext` this way and override it. Feels clunky. but not sure how else to do it without creating a big mess everywhere else.

Comment (scoquelin) [reply modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisCommandDelegation.scala:28]: Yeah that's what I was looking into quickly... could not come up with anything more elegant. We would need [trait parameters](https://docs.scala-lang.org/scala3/reference/other-new-features/trait-parameters.html) maybe from Scala 3

File: modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala (hunk 2 above)
Comment (scoquelin) [reply modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala:1]: Yes I think that would be better to put everything in the `com.github.scoquelin.arugula.commands` but I think the lower level client implementations should be moved inside `client/internal` if that's even possible... we can adjust this in a separate PR though

**Review by scoquelin — APPROVED** [review]
Nice work! Makes it way better to isolate/group by Redis functions ! Thanks! 🚀

File: modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala (hunk 2 above)
Comment (72squared) [reply modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisBaseAsyncCommands.scala:1]: ok. that can be in a follow-on PR

**Pull request merged**